# Estudo de Caso - RESOLUÇÃO!
Por Amanda, Marcela e Maria.

# Importando as bibliotecas.


In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Carregamento de dados.


In [ ]:
df_maquinas = pd.read_json('/content/dados_maquinas.json')
df_insumos = pd.read_json('/content/dados_insumos.json')
df_turnos = pd.read_json('/content/dados_turnos.json')

# Respondendo as perguntas com Python.


# Pergunta 1: **Qual máquina causou o problema?**

Vamos verificar **status**, **alerta** e **histórico** das máquinas.

In [ ]:
df_maquinas[["id", "nome", "status", "alerta"]]

id             nome status           alerta
0   1   MP-01: Picador     ok             None
1   2  MP-02: Digestor   warn  Pressão Crítica
2   3   MP-03: Secagem     ok             None

Resultado esperado:

* MP-02 (Digestor)
* `status = warn`
* `alerta = Pressão Crítica`

Agora vamos olhar o **último registro do histórico** da MP-02:

In [ ]:
mp02 = df_maquinas[df_maquinas["id"] == 2].iloc[0]
pd.DataFrame(mp02["historico"])

data  producao  temp   status
0  2025-07-01       320   160   Normal
1  2025-08-15       315   162   Normal
2  2025-09-10       330   165  Atenção
3  2025-11-20       340   170  Atenção
4  2025-12-01       350   172  Atenção
5  2025-12-15       355   174  Crítico
6  2026-01-10       100   175    FALHA

Observação:

* Temperatura chegou a **175°C**
* Status final: **FALHA**
* Produção caiu drasticamente (100)

Conclusão 1:

**A máquina culpada é a MP-02 (Digestor).**

# Pergunta 2: **Por que a temperatura subiu tanto?**

Vamos analisar os **insumos**, especialmente a **Soda Cáustica**.

In [ ]:
df_insumos[["nome", "nivel", "st"]]

nome      nivel       st
0  Soda Cáustica        22%  CRÍTICO
1    Resina Imp.        10%  CRÍTICO
2        Madeira  12.500 m³   Normal

Agora filtrar apenas a Soda Cáustica:

In [ ]:
soda = df_insumos[df_insumos["nome"] == "Soda Cáustica"].iloc[0]
pd.DataFrame(soda["historico"])

data  pct
0  2025-07-01   95
1  2025-08-01   80
2  2025-09-01   65
3  2025-10-01   50
4  2025-11-01   35
5  2025-12-01   28
6  2026-01-10   22

Evolução clara:

* 95% → 80% → 65% → 50% → 35% → 28% → **22%**
* Status: **CRÍTICO**

**Lógica de processo industrial**:

* Pouca soda = reação química fraca
* Para compensar → **aumenta temperatura e pressão**
* Resultado → **papel queimado**

**Conclusão 2:**:

A temperatura subiu porque o **nível crítico de Soda Cáustica (22%) forçou o aumento térmico no Digestor**.

# Pergunta 3: **Quem estava operando a máquina no momento?**

O lote problemático foi produzido:


* **10/01**

* **Turno: Tarde**

* **Máquina: MP-02**


Vamos filtrar isso:

In [ ]:
df_turnos[
    (df_turnos["data"] == "10/01") &
    (df_turnos["turno"] == "Tarde") &
    (df_turnos["maq"] == 2)
]

data  turno  maq          op            obs
1  10/01  Tarde    2  Mariana S.  Falha Pressão

Resultado:
* Operadora: **Mariana S.**
* Observação: **Falha de Pressão**
* **Conclusão 3:**
Quem operava a máquina era **Mariana S. (Trainee)**.